src/rust/en.mangairo/src/parser.rs

```diff
@@ -0,0 +1,354 @@
+use aidoku::{
+	error::Result, helpers::uri::QueryParameters, prelude::*, std::html::Node, std::String,
+	std::Vec, Chapter, Filter, FilterType, Manga, MangaContentRating, MangaStatus, MangaViewer,
+	Page,
+};
+extern crate alloc;
+use alloc::string::ToString;
+
+pub const BASE_URL: &str = "https://w.mangairo.com";
+pub const USER_AGENT: &str = "Mozilla/5.0 (Macintosh; Intel Mac OS X 13_3_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36";
+
+pub fn parse_manga_list(html: Node, page: i32) -> (Vec<Manga>, bool) {
+	let mut result: Vec<Manga> = Vec::new();
+	for page in html.select(".story-item").array() {
+		let obj = page.as_node().expect("node array");
+
+		let id = obj.select(".story-name a").attr("href").read();
+		let title = obj.select(".story-name a ").text().read();
+		let cover = obj.select(".story-list-img img").attr("src").read();
+
+		if !id.is_empty() && !title.is_empty() && !cover.is_empty() {
+			result.push(Manga {
+				id,
+				cover,
+				title,
+				..Default::default()
+			});
+		}
+	}
+
+	// Example: 'Total: 38,202 stories'
+	let mut total_str = html.select(".quantitychapter").text().read();
+	total_str = total_str.replace("Total: ", "");
+	total_str = total_str.replace(" stories", "");
+	total_str = total_str.chars().filter(|&c| c != ',').collect();
```

Comment:
this doesn't need to be mutable, you could do all the calls on one variable (including the i32 parse)